libafl/src/stages/tuneable.rs

```diff
@@ -175,16 +175,14 @@ where
         manager: &mut EM,
         corpus_idx: CorpusId,
     ) -> Result<(), Error> {
-        let metadata: &TuneableMutationalStageMetadata = state.metadata()?;
+        let fuzz_time = self.seed_fuzz_time(state)?;
+        let iters = self.iters(state)?;
 
-        let fuzz_time = metadata.fuzz_time;
-        let iters = metadata.iters;
-
-        let (start_time, iters) = if fuzz_time.is_some() {
-            (Some(current_time()), iters)
-        } else {
-            (None, Some(self.iterations(state, corpus_idx)?))
-        };
+        if fuzz_time.is_some() && iters.is_some() {
+            return Err(Error::illegal_state(
```

Comment:
Why would that not be legal?

Reply:
It doesn't make sense to specify both; you may wish to fuzz for a particular amount of time, or you may want to fuzz for a certain number of iterations. I suppose it's possible to want to fuzz for a particular amount of iterations, but also restrict the amount of time, but the way that the code was previously written didn't actually account for this and just assumed that only one was set.

Reply:
Ideally it would do exit on whatever is reached first, right?